content/en/apps/guides/hosting/monitoring.md

```diff
@@ -79,3 +79,161 @@ Messaging Outgoing State Failed | `non-zero` | 0
+
+These instructions have been tested against Ubuntu, but should work against any OS that meets the prerequisites. They follow a happy path assuming you need to only set a secure password and specify the URL(s) to monitor:
+
+1. Run the following commands to clone this repository, initialize your `.env` file, create a secure password and create your data directories:
+
+    ```sh
+    cd ~
+    git clone https://github.com/medic/cht-monitoring.git
+    cd cht-monitoring
+    cp cht-instances.example.yml cht-instances.yml
+    cp grafana/grafana.example.ini grafana/grafana.ini
+    mkdir -p grafana/data && mkdir  -p prometheus/data 
+    apt install -y wamerican
+    cp .env.example .env
+    password=$(shuf -n7 /usr/share/dict/words --random-source=/dev/random | tr '\n' '-' | tr -d "'" | cut -d'-' -f1,2,3,4,5,6,7)
+    sed -i -e "s/password/$password/g" .env
+    echo;echo "Setup complete! To log into Grafana in the browser:";echo 
+    echo "    username: medic"
+    echo "    password: ${password}";echo
+    ```
+   Note that in step 4 below you'll need the username and password which is printed after you run the above command.
+
+2. Edit the `cht-instances.yml` file to have the URLs of your CHT instances. You may include as many URLs of CHT instances as you like. To preserve data consistency parameters in this file should NOT be adjusted after it is initially set. Be sure to not include a trailing slash (`/`).
+
+   Here is an example:
+
+    ```yml
+    - targets:
+      - https://subsub.sub.example.com
+      - https://cht.domain.com
+      - https://website.org
+    ```
+3. Run the following command to deploy the stack:
+    
+    ```sh
+    cd ~/cht-monitoring
+    docker compose up -d
+    ```
+
+4. Grafana is available at [http://localhost:3000](http://localhost:3000). See the output from step 1 for your username and password.
+
+If you would like to do more customizing of your deployment, see ["Addition Configuration"](#additional-configuration).
+
+### Upgrading
+
+Before upgrading, you should back up both your current configuration settings and your Prometheus time-series data.
+
+#### Prometheus, Grafana and JSON Exporter
+
+To upgrade these dependencies, update the version numbers set in your `.env` file (or leave them set to `latest`).  Then run the following commands:
+
+```shell
+docker compose pull
+docker compose up -d
+```
+
+#### CHT Monitoring Config
+
+When you see a new version in the [GitHub repository](https://github.com/medic/cht-monitoring), first review the release notes and upgrade instructions. Then, run the following commands to deploy the new configuration:
+
+1. Note the version number to derive the branch to pull.  For example,  `1.1.0` would be `1.1.x`
+2. Run this command to update your instance and restart it. Be sure to replace `BRANCH_NAME` with the value from the first step:
+   ```shell
+   cd ~/cht-monitoring
+   git fetch
+   git checkout BRANCH_NAME
+   ```shell
```

Comment:
Since new versions of the config could include additions to the docker-compose file, I think this should include:

```shell
docker compose pull
docker compose up -d --remove-orphans
```

Reply:
ah - ok - I had removed that thinking we might not want to upgrade grafana/prometheus when we're just trying to update the cht-monitoring part.  I guess that should come in a separate commit to pin to a specific release vs `latest` like we have no.  That works!

I added a restart in 659ace5 and then I added `pull` just now in 1b34f0a.  Lemme know if we require the `compose up -d --remove-orphans` , assuming not though!

Reply:
So, I might be wrong, but I don't think `restart` will actually update the image used (even if you have done a `pull` and gotten a new version) if you have existing containers (it just keeps the same containers).

The case I am trying to cover is if we ever update our base docker-compose config to add or remove a service (e.g. removing json-exporter since we don't need it any more).  In that case we also want the `--remove-orphans` since the json-exporter service will not be in the new version of the docker-compose file.

Reply:
gotcha!  Added `--remove-orphans` to both calls and added a `docker compose down` before the `up` to not have the `restart` any more in e37e1be4.  Thanks for seeing this through!